xarray/core/dataarray.py

```diff
@@ -3234,6 +3234,43 @@ def map_blocks(
 
         return map_blocks(func, self, args, kwargs)
 
+    def polyfit(
+        self,
+        dim: Hashable,
+        deg: int,
+        skipna: bool = None,
+        rcond: float = None,
+        w: Union[Hashable, Any] = None,
+        full: bool = False,
+        cov: bool = False,
+    ):
+        """
+        Least squares polynomial fit.
+
+        This replicates the behaviour of `numpy.polyfit` but differs by skipping
+        invalid values when `skipna = True`.
+
+        Parameters
+        ----------
+        dim : hashable
+            Coordinate along which to fit the polynomials.
+        skipna : bool, optional
+            If True, removes all invalid values before fitting each 1D slices of the array
+            Defaults is True if data is stored in as dask.array or if there is any
+            invalid values, False otherwise.
+
+        See documentation of `numpy.polyfit`.
+
+        Returns
+        -------
+        A single dataset with the coefficients of the best fit for this DataArray.
+        The residuals, singular values and rank are included if `full` is True.
+        The covariance matrix is included if `full` is False and `cov` is True.
```

Comment:
Might be clearer to say it returns a dataset which contains the variables:
- polyfit_coefficients : The coefficients which...
- polyfit_residuals : The residuals...  (only included if ...)
- polyfit_covariance : The residuals...  (only included if ...)